client/components/select-vertical/suggestion-search.tsx

```diff
@@ -0,0 +1,118 @@
+import { Suggestions } from '@automattic/components';
+import classnames from 'classnames';
+import { useTranslate } from 'i18n-calypso';
+import { useCallback, useMemo, useRef, useState } from 'react';
+import * as React from 'react';
+import FormTextInput from 'calypso/components/forms/form-text-input';
+import type { SiteVerticalsVertical } from 'calypso/data/site-verticals';
+import './style.scss';
+
+interface Props {
+	placeholder?: string;
+	searchTerm: string;
+	suggestions: SiteVerticalsVertical[];
+	autoFocus: boolean;
+	isLoading?: boolean | undefined;
+	onInputChange?: ( value: string ) => void;
+}
+
+const SelectVerticalSuggestionSearch: React.FC< Props > = ( {
+	placeholder,
+	searchTerm,
+	suggestions,
+	autoFocus,
+	isLoading,
+	onInputChange,
+} ) => {
+	const [ isShowSuggestions, setIsShowSuggestions ] = useState( false );
+	const [ isFocused, setIsFocused ] = useState( false );
+	const suggestionsRef = useRef( null );
+	const translate = useTranslate();
+
+	const handleTextInputBlur = useCallback( () => {
+		setIsFocused( false );
+	}, [ setIsFocused ] );
+
+	const handleTextInputFocus = useCallback( () => {
+		if ( 0 < suggestions.length ) {
+			setIsShowSuggestions( true );
+		}
+
+		setIsFocused( true );
+	}, [ suggestions, setIsFocused, setIsShowSuggestions ] );
+
+	const handleTextInputChange = useCallback(
+		( event: React.ChangeEvent< HTMLInputElement > ) => {
+			setIsShowSuggestions( 0 < event.target.value.trim().length );
+			onInputChange?.( event.target.value );
+		},
+		[ setIsShowSuggestions, onInputChange ]
+	);
+
+	const handleTextInputKeyDown = useCallback(
+		( event: KeyboardEvent ) => {
+			if ( event.key === 'Enter' ) {
+				event.preventDefault();
+			}
+
+			if ( suggestionsRef.current ) {
+				( suggestionsRef.current as Suggestions ).handleKeyEvent( event );
+			}
+		},
+		[ suggestionsRef ]
+	);
+
+	const handleSuggestionsSelect = useCallback(
+		( { label }: { label: string } ) => {
+			setIsShowSuggestions( false );
+			onInputChange?.( label );
+		},
+		[ setIsShowSuggestions, onInputChange ]
+	);
+
+	const getSuggestions = useMemo( () => {
+		if ( isLoading || ! isShowSuggestions ) {
+			return [];
+		}
+
+		return suggestions.concat( [
+			{
+				value: '',
+				label: String( translate( 'Something else' ) ),
```

Comment:
I think “Something else“ is weird. I think we should not be able to select this one. Do you have the spec of this? What I imagine is that it should be the same as the input text.

Reply:
According to the Figma prototype (pbxlJb-1nn-p2) this should be clickable, but I'm sure this is something we'll iterate on.